### PR TITLE
FEATURE: Allow to modify topic-backed static pages

### PR DESCRIPTION
**Pull request by CvX**
A plugin API that allows customizing existing topic-backed static pages, like:
faq, tos, privacy (see: StaticController)

```
add_topic_static_page("faq") do |controller|
  current_user&.locale == "pl" ? "polish_faq_topic_id" : "faq_topic_id"
end
```

You can also add new pages in a plugin, but remember to add a route,
for example:

```
get "contact" => "static#show", id: "contact"
```

### CODE REVIEW

**Comment by CvX** [comment]
@tgxworld @pmusaraj That's the different approach for locale-specific static pages in plugins. What do you think?

**Review by pmusaraj — APPROVED** [review]
Looks good to me!

File: lib/plugin/instance.rb [hunk 1]
```diff
@@ -994,6 +994,21 @@ def register_notification_consolidation_plan(plan)
     DiscoursePluginRegistry.register_notification_consolidation_plan(plan, self)
   end
 
+  # Allows customizing existing topic-backed static pages, like:
+  # faq, tos, privacy (see: StaticController)
+  #
+  #   add_topic_static_page("faq") do |controller|
+  #     current_user&.locale == "pl" ? "polish_faq_topic_id" : "faq_topic_id"
+  #   end
```
Comment (tgxworld) [inline lib/plugin/instance.rb:1002]: This looks good to me :+1: One suggestion though is that I think we need to explain here what the return value is supposed to be. In this case, the return value is the name of the `SiteSetting`.

**Review by tgxworld — APPROVED** [review]
Looks good to me :+1: Thank you for making this change.

**Pull request merged**